src/embeddings.rs

```diff
@@ -180,17 +181,50 @@ impl PyEmbeddings {
         Self::similarity_results(py, results)
     }
 
+    /// embedding(word,/, default)
+    /// --
+    ///
     /// Get the embedding for the given word.
     ///
     /// If the word is not known, its representation is approximated
-    /// using subword units.
-    fn embedding(&self, word: &str) -> Option<Py<PyArray1<f32>>> {
+    /// using subword units. #
+    ///
+    /// If no representation can be calculated:
+    ///  - `None` if `default` is `None`
+    ///  - an array filled with `default` if `default` is a scalar
+    ///  - an array if `default` is a 1-d array
+    ///  - an array filled with values from `default` if it is an iterator over floats.
+    #[args(default = "PyEmbeddingDefault::default()")]
+    fn embedding(
+        &self,
+        word: &str,
+        default: PyEmbeddingDefault,
+    ) -> PyResult<Option<Py<PyArray1<f32>>>> {
         let embeddings = self.embeddings.borrow();
 
-        embeddings.embedding(word).map(|e| {
-            let gil = pyo3::Python::acquire_gil();
-            e.into_owned().into_pyarray(gil.python()).to_owned()
-        })
+        let gil = pyo3::Python::acquire_gil();
+        if let Some(embedding) = embeddings.embedding(word) {
+            return Ok(Some(
+                embedding.into_owned().into_pyarray(gil.python()).to_owned(),
+            ));
+        };
+        match default {
+            PyEmbeddingDefault::Constant(constant) => {
+                let nd_arr = Array1::from_elem([embeddings.storage().shape().1], constant);
+                Ok(Some(nd_arr.into_pyarray(gil.python()).to_owned()))
+            }
+            PyEmbeddingDefault::Embedding(array) => {
+                if array.as_ref(gil.python()).shape()[0] != embeddings.storage().shape().1 {
```

Comment:
I wonder if this check should move to the beginning. It would be nice if this is an error before you hit this weird token for which no embedding could be computed.

Reply:
Not sure if I understand correctly: Do you always want to match `default` and verify its shape?

Reply:
Yes. I don't know what the cost is. But it would check the invariants early.

Reply:
Ah, I see that is what you are doing now ;).